src/Components/ImagesTable/Target.tsx

```diff
@@ -0,0 +1,54 @@
+import {
+  useGetClonesQuery,
+  useGetComposeStatusQuery,
+} from '../../store/apiSlice';
+
+type TargetProps = {
+  composeId: string;
+};
+
+const targetOptions = {
+  aws: 'Amazon Web Services',
+  'aws.s3': 'Amazon Web Services',
+  azure: 'Microsoft Azure',
+  gcp: 'Google Cloud Platform',
+  vsphere: 'VMWare',
+  'guest-image': 'Virtualization - Guest image',
+  'image-installer': 'Bare metal - Installer',
+  'edge-commit': 'edge-commit',
+  'edge-installer': 'edge-installer',
+  ami: 'ami',
+  'rhel-edge-commit': 'rhel-edge-commit',
+  'rhel-edge-installer': 'rhel-edge-installer',
+  vhd: 'vhd',
+};
+
+const Target = ({ composeId }: TargetProps) => {
+  const { data: clones, isSuccess: isSuccessClones } =
+    useGetClonesQuery(composeId);
+  const { data: composeStatus, isSuccess: isSuccessTarget } =
+    useGetComposeStatusQuery(composeId);
+  const target = composeStatus?.request?.image_requests[0];
+
+  let tar;
+
+  if (isSuccessTarget) {
```

Comment:
This is a change in behaviour, we didn't check if the target succeeded before. Why is this necessary?

Reply:
Hmm, this is not really checking for the success of the target, rather for the success of fetching the compose information as the `useSelector()` hook is replaced by fetching the data via RTKQuery hook. This ensures we are not trying to read an image type from a request that is not resolved yet, ending up with undefined type.

A change in the name of the variable might be in place from `isSuccessTarget` to `isSuccessComposeStatus`

Reply:
change the name to isSuccessComposeStatus